lib/main.dart

```diff
@@ -0,0 +1,187 @@
+import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+void main()=> runApp(const MyApp());
+
+class Movie {
+
+  final String title;
+  final String originalTitle;
+  final String overview;
+  final DateTime releaseDate;
+  final double voteAverage;
+  final List<String> genres;
+  final String poster;
+  final String backdrop;
+  int likes=0;
+
+  Movie( this.title, this.originalTitle, this.overview, this.releaseDate,
+      this.voteAverage, this.genres, this.poster,this.backdrop);
+
+  void likeMovie() => likes++;
+
+  String showGenres(){
+    String output='';
+    for (var element in genres) {
+      output='$output  $element';
+    }
+    return output;
+  }
+}
+
+ThemeData _buildTheme(brightness) {
+  final baseTheme = ThemeData(brightness: brightness);
+
+  return baseTheme.copyWith(
+    textTheme: GoogleFonts.latoTextTheme(baseTheme.textTheme),
+  );
+}
+
+class LikeButton extends StatefulWidget{
+  final Movie movie;
+  const LikeButton(this.movie, {super.key});
+
+
+  @override
+  State<StatefulWidget> createState() => _LikeButton();
+
+}
+
+class _LikeButton extends State<LikeButton>{
+
+  static const IconData likeIcon=Icons.favorite;
+
+  void addLike(Movie movie){
+    movie.likeMovie();
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+
+    return
+      ElevatedButton.icon(
+        onPressed: (){
+          addLike(widget.movie);
+        },
+        icon: const Icon(likeIcon),
+        label: Text('${widget.movie.likes}'),
+      );
+  }
+}
+
+class MyApp extends StatelessWidget{
+
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return
+      MaterialApp(
+          home: MyHomePage()
+      );
+  }
+}
+
+
+class MyHomePage extends StatelessWidget {
+  MyHomePage({super.key});
+
+  static const Color backgroundColor = Color.fromRGBO(255, 255, 255, 0.8);
+
+  static const double appBarToolbarHeight=50;
+
+  static const double posterWidth =200;
+
+  static const double largePadding=10;
+  static const double mediumPadding=5;
+  static const double smallPadding=2.5;
+
+  static const Icon appBarBackIcon = Icon(Icons.arrow_back);
+
+  static const double largeFont = 20;
+  static const double mediumFont = 15;
+
+  final Movie movie=Movie("The Super Mario Bros. Movie",
```

Comment:
We can have a constructor in the movie class to create an instance with this values

Reply:
fixed